src/main/java/org/mockito/internal/invocation/InvocationMatcher.java

```diff
@@ -134,24 +133,12 @@ private void captureVarargsPart(Invocation invocation) {
             return;
         }
 
-        Object[] rawArguments = invocation.getRawArguments();
-        int indexOfVarArg = rawArguments.length - 1;
+        Object[] arguments = invocation.getArguments();
 
-        Object varArg = rawArguments[indexOfVarArg];
-        if (varArg == null) {
-            return;
-        }
-        int varArgSize = Array.getLength(varArg);
-
-        if (matchers.size() == indexOfVarArg) {
-            return;
-        }
-
-        for (int i = 0; i < varArgSize; i++) {
-            int indexOfVarargMatcher = (i + indexOfVarArg > matchers.size() -1 ? indexOfVarArg : i + indexOfVarArg);
-            ArgumentMatcher m = matchers.get(indexOfVarargMatcher);
+        for (int i = 0; i < arguments.length; i++) {
+            ArgumentMatcher m = matchers.get(i > matchers.size() - 1 ? matchers.size() - 1 : i);
             if (m instanceof CapturesArguments) {
-                ((CapturesArguments) m).captureFrom(Array.get(varArg, i));
+                ((CapturesArguments) m).captureFrom(arguments[i]);
             }
         }
```

Comment:
@terebesirobert  Hi Robert, 
I'm not sure what would break the backward compatibility. Do you mean this case: `mock.varArgs((String[]) null)`  _<- should capture nothing_ ?

```
if (args[nonVarArgsCount] == null) {
            // in case someone deliberately passed null varArg array
            varArgs = new Object[0]; // an empty array should do the trick...
```